File: typescript/src/resources/tiktok.ts
/** TikTok data resources (`/tiktok/*`). */

import { Transport } from '../runtime/transport';

export class Tiktok {
  constructor(private transport: Transport) {}

  posts(opts: Record<string, unknown> = {}): Promise<Record<string, unknown>> {
    return this.transport.request('POST', '/tiktok/posts', { json: opts });
  }

  search(opts: { type: 'user' | 'video'; keywords: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    return this.transport.request('POST', '/tiktok/search', { json: opts });
  }

  user(opts: { type: string; keywords: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    return this.transport.request('POST', '/tiktok/user', { json: opts });
  }

  video(opts: { videoUrl: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { videoUrl, ...rest } = opts;
    return this.transport.request('POST', '/tiktok/video', { json: { video_url: videoUrl, ...rest } });
  }
}

File: python/src/acedatacloud/resources/tiktok.py
"""TikTok data resources (``/tiktok/*``)."""

from __future__ import annotations

from typing import Any, Literal


class Tiktok:
    """Synchronous TikTok client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def posts(self, **kwargs: Any) -> dict[str, Any]:
        return self._transport.request("POST", "/tiktok/posts", json=kwargs)

    def search(
        self,
        *,
        type: Literal["user", "video"],
        keywords: str,
        **kwargs: Any,
    ) -> dict[str, Any]:
        return self._transport.request("POST", "/tiktok/search", json={"type": type, "keywords": keywords, **kwargs})

    def user(self, *, type: str, keywords: str, **kwargs: Any) -> dict[str, Any]:
        return self._transport.request("POST", "/tiktok/user", json={"type": type, "keywords": keywords, **kwargs})

    def video(self, *, video_url: str, **kwargs: Any) -> dict[str, Any]:
        return self._transport.request("POST", "/tiktok/video", json={"video_url": video_url, **kwargs})


class AsyncTiktok:
    """Async TikTok client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def posts(self, **kwargs: Any) -> dict[str, Any]:
        return await self._transport.request("POST", "/tiktok/posts", json=kwargs)

    async def search(self, *, type: Literal["user", "video"], keywords: str, **kwargs: Any) -> dict[str, Any]:
        return await self._transport.request(
            "POST", "/tiktok/search", json={"type": type, "keywords": keywords, **kwargs}
        )

    async def user(self, *, type: str, keywords: str, **kwargs: Any) -> dict[str, Any]:
        return await self._transport.request(
            "POST", "/tiktok/user", json={"type": type, "keywords": keywords, **kwargs}
        )

    async def video(self, *, video_url: str, **kwargs: Any) -> dict[str, Any]:
        return await self._transport.request("POST", "/tiktok/video", json={"video_url": video_url, **kwargs})
